extensions/vscode/src/state.ts

```diff
@@ -283,11 +287,30 @@ export class PublisherState implements Disposable {
         this.credentials = response.data;
       });
     } catch (error: unknown) {
+      if (isErrCredentialsCorrupted(error)) {
+        this.resetCredentials();
+        return;
+      }
       const summary = getSummaryStringFromError("refreshCredentials", error);
       window.showErrorMessage(summary);
     }
   }
 
+  // Calls to reset all credentials data stored.
+  // Meant to be a last resort when we get loading data or corrupted data errors.
+  async resetCredentials() {
+    try {
+      const api = await useApi();
+      const response = await api.credentials.reset();
+      this.credentials = [];
```

Comment:
While it is logical to think that reset would simply result in empty credentials, I think it would be better to follow our existing patterns where we rely on the business logic to be implemented within the agent API. So in this case, we'd change the API result to return the list of credentials, which would be empty. That would be the value that would be updated within the extension.

Reply:
I went back and forth on this comment for a while wondering if it was nitpicky, and a bit silly (since we know it is going to be `[]`) but decided to implement it. My reasoning was, if we add any features in the future to allow setting other credentials that the `reset` doesn't cover (think reading the environment variables or something) then this will be the more thorough code path.